lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -150,106 +133,255 @@ extension ASTGenVisitor {
 // Misc visits.
 // TODO: Some of these are called within a single file/method; we may want to move them to the respective files.
 extension ASTGenVisitor {
-  public func visit(_ node: MemberBlockItemSyntax) -> ASTNode {
-    visit(Syntax(node.decl))
+  public func generate(_ node: MemberBlockItemSyntax) -> ASTNode {
+    generate(node.decl)
   }
 
-  public func visit(_ node: InitializerClauseSyntax) -> ASTNode {
-    visit(node.value)
+  public func generate(_ node: InitializerClauseSyntax) -> ASTNode {
+    generate(node.value)
   }
 
-  public func visit(_ node: ConditionElementSyntax) -> ASTNode {
-    visit(node.condition)
+  public func generate(_ node: ConditionElementSyntax) -> ASTNode {
+    generate(node.condition)
+  }
+
+  public func generate(_ node: CodeBlockItemSyntax) -> ASTNode {
+    generate(node.item)
+  }
+
+  public func generate(_ node: ArrayElementSyntax) -> ASTNode {
+    generate(node.expression)
+  }
+    
+  func generate(_ node: DeclSyntax) -> ASTNode {
+      return generate(Syntax(node))
   }
 
-  public func visit(_ node: CodeBlockItemSyntax) -> ASTNode {
-    visit(node.item)
+  func generate(_ node: ExprSyntax) -> ASTNode {
+    return generate(Syntax(node))
   }
 
-  public func visit(_ node: ArrayElementSyntax) -> ASTNode {
-    visit(node.expression)
+  func generate(_ node: PatternSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: StmtSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: TypeSyntax) -> ASTNode {
+    return generate(Syntax(node))
+  }
+
+  func generate(_ node: some SyntaxChildChoices) -> ASTNode {
+    return self.generate(Syntax(node))
   }
 
   @inline(__always)
-  func visit(_ node: CodeBlockItemListSyntax) -> BridgedArrayRef {
-    node.lazy.map { self.visit($0).bridged }.bridgedArray(in: self)
+  func generate(_ node: CodeBlockItemListSyntax) -> BridgedArrayRef {
+    node.lazy.map { self.generate($0).bridged }.bridgedArray(in: self)
+  }
+    
+  public func generate(_ node: Syntax) -> ASTNode {
+    switch node.as(SyntaxEnum.self) {
+    case .memberBlockItem(let node):
+        return generate(node)
+    case .initializerClause(let node):
+        return generate(node)
+    case .conditionElement(let node):
+        return generate(node)
+    case .codeBlockItem(let node):
+        return generate(node)
+    case .arrayElement(let node):
+        return generate(node)
+    case .typeAliasDecl(let node):
+        return generate(node)
+    case .enumDecl(let node):
+        return generate(node)
+    case .structDecl(let node):
+        return generate(node)
```

Comment:
Could you make sure to indent `return` by two spaces to match the style of the project?

Also, could you sort the case names alphabetically. In my experience, that’s cleaner than random ordering.